cns/service/main.go

```diff
@@ -857,10 +857,22 @@ func reconcileInitialCNSState(ctx context.Context, cli nodeNetworkConfigGetter,
 
 	// Convert to CreateNetworkContainerRequest
 	for i := range nnc.Status.NetworkContainers {
-		ncRequest, err := kubecontroller.CreateNCRequestFromDynamicNC(nnc.Status.NetworkContainers[i])
+		var ncRequest *cns.CreateNetworkContainerRequest
+		var err error
+		switch nnc.Status.NetworkContainers[i].AssignmentMode {
+		case v1alpha.Static:
+			ncRequest, err = kubecontroller.CreateNCRequestFromStaticNC(nnc.Status.NetworkContainers[i])
+		case v1alpha.Dynamic:
+		default: // For backward compatibility, default will be treated as Dynamic too.
+			ncRequest, err = kubecontroller.CreateNCRequestFromDynamicNC(nnc.Status.NetworkContainers[i])
```

Comment:
Go switches do not fall-through like this. It would be better to remove the case `Dynamic` case entirely for now. You will need to add a `//nolint:exhaustive // skipping current default case.` to satisfy golint